test/integration/puppetlabs/services/jruby/jruby_locking_test.clj

```diff
@@ -15,7 +15,16 @@
 (defn jruby-service-test-config
   [pool-size]
   (jruby-testutils/jruby-puppet-tk-config
-    (jruby-testutils/jruby-puppet-config {:max-active-instances pool-size})))
+    (jruby-testutils/jruby-puppet-config {:max-active-instances pool-size
+                                          :borrow-timeout 1})))
```

Comment:
If an instance is available in the queue and the timeout is anything greater than 0, the instance will be returned.  The wait timer doesn't start unless/until it is determined that no instance is immediately available to be borrowed.